omod/src/main/java/org/openmrs/module/queue/web/QueueEntryTransitionRestController.java

```diff
@@ -96,6 +101,36 @@ public Object transitionQueueEntry(@RequestBody QueueEntryTransitionRequest body
 		
 		transition.setNewPriorityComment(body.getNewPriorityComment());
 		
+		// Location waiting for
+		if (rawBody.containsKey(QueueEntryTransitionRequest.NEW_LOCATION_WAITING_FOR_FIELD)) {
```

Comment:
Was there a reason these couldn't just be added as properties for the QueueEntryTransitionRequest?

Reply:
They are properties already, but having those properties be null is not sufficient to distinguish between "we intend to set this to null" vs "we don't want to change this property". So we check the rawBody to see if those optional properties are provided at all, and only set the corresponding queue entry field if so.

Reply:
@chibongho - isn't that also the case for some of the other, already existing properties you are setting?  For example `priorityComment`?

Reply:
@mseaton that's a good point about `priorityComment` (and other nullable fields). We got away with it because the frontend defaults the `QueueEntryTransitionRequest.newPriorityComment` field to whatever the original queue entry's `priorityComment` is, or empty string if its null. The UI renders null or empty string the same way anyway.

I originally intended the `QueueEntryTransitionRequest` to have all fields be optional and the request would only modify the fields that are provided. That seems to not work as well for nullable fields. I can just change the API so that all fields are required, and the request would have to populate all non-modified fields with values from the original queue entry. (We already do that in the fronend).